File: 2870-minimum-number-of-operations-to-make-array-empty/2870-minimum-number-of-operations-to-make-array-empty.java
class Solution {
    public int minOperations(int[] nums) {
        HashMap<Integer, Integer> mp = new HashMap<> ();
        for(int i=0; i<nums.length; i++){
            int x= mp.getOrDefault(nums[i], 0);
            mp.put(nums[i], x+1);
        }
        int ans=0;
        for(Map.Entry m: mp.entrySet()){
            int val = (Integer)m.getValue();
            if(val==1) return -1;
            ans+= (val/3) + ((val%3>0)?1:0);
        }
        return ans;
    }
}

File: 2870-minimum-number-of-operations-to-make-array-empty/2870-minimum-number-of-operations-to-make-array-empty.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        unordered_map<int,int> mp;
        for(auto &i: nums) mp[i]++;
        int ans=0;
        for(auto &i: mp){
            int val= i.second;
            if(val==1) return -1;
            ans+= (val/3);
            if(val%3!=0) ans++;
        }
        return ans;
    }
};